src/inet/http_server.c

```diff
@@ -196,10 +196,16 @@ find_route_callback(struct http_server_t *self_p,
 {
     const struct http_server_route_t *route_p;
 
+    int path_length = strlen(path_p);
```

Comment:
Nice change. I just have a coding style comment. Preferably declare variables at the beginning of the function and modify them later.

```
    int path_length;
    char *query_string_p;

    /* The query string is not part of the route. */
    query_string_p = strstr(path_p, "?");
    
    if (query_string_p != NULL) {
        path_length = (query_string_p - path_p);
    } else {
        path_length = strlen(path_p);
    }

```